### DISCUSSION TITLE
How to improve performance?

### DISCUSSION THREAD

**Discussion by ChillarAnand** [marlonrichert/zsh-autocomplete#511] [General]
When I started using zsh-autocomplete, live history search wasn't working. I spent couple of hours to debug and finally realised live auto complete is slow because of my history.

I have been saving/syncing up shell history from ~10 years. Now it has ~0.25 million entries. I did some clean up and reduced the entries to < 0.1million. Still autocomplete is slow.

Any suggestions on improving the search performance?

**Comment by marlonrichert** [comment]
There is no workaround at the moment, other than reducing history size. There is an open issue to address this problem: #459
